### DISCUSSION TITLE
Integrazione telecamera, errore sicuramente stupido

### DISCUSSION THREAD

**Discussion by JDrillz** [indomus/forum#2954] [HUB: Home Assistant]
Ciao a tutti, sto da pochissimo smanettando con home assistant, e dovrei integrarci una telecamera Arlo, ma avendo 0 esperienza con questo hub sto sicuramente facendo un errore stupido, perché la guida è chiarissima ma non funziona niente.

Ho installato HACS, ho scaricato Arlo Camera Support ed ho aggiunto al configuration.yaml questa configurazione base base copia/incollata:

```
aarlo:
  username: !secret arlo_username
  password: !secret arlo_password
```

Ho sostituito ovviamente arlo_username e arlo_password con le mie credenziali e salvato. Però sulla panoramica non compare assolutamente niente.
Ho provato sia inserendo le credenziali del mio account principale, sia quelle di un account specifico HA cui ho condiviso la visualizzazione della telecamera (come suggerito di fare). Ma in nessuno dei due casi visualizzo niente.

La guida sembrerebbe banale da seguire, ma evidentemente faccio qualche errore da novellino che non riesco a comprendere e che la guida da per scontato.

Cosa sbaglio? Devo fare altro per vedere qualcosa? Grazie.

**Comment by M0ntyC0r4** [comment]
credo tu debba dargli anche, come minimo, "host: http://...."
In rete se può servire ho trovato [questo](https://www.reddit.com/media?url=https%3A%2F%2Fpreview.redd.it%2Farlo-setup-v0-gb5wrw4wvfsa1.png%3Fwidth%3D490%26format%3Dpng%26auto%3Dwebp%26s%3D8fe267ac422a22f2bbde52ceb76aac65f13948d8) e [questo](https://community.home-assistant.io/t/solved-odd-errors-during-aarlo-setup/583088) ma non avendo telecamere Arlo mi fermo qui...

**Reply by JDrillz** [reply]
Intanto grazie per l'aiuto, nel frattempo ho fatto qualche progresso. Mi ero perso una nota di aggiornamento che diceva che adesso il 2FA è obbligatorio, quindi l'ho attivato. Tuttavia pur non avendo errori nello yaml continuo a non vedere niente sulla plancia, come se non trovasse la telecamera.
Il login lo effettua correttamente, infatti al primo riavvio post-configurazione mi è arrivata una mail del 2FA e mi ha sloggato dall'accesso che avevo effettuato dicendomi che qualcun altro (HA) si era collegato.
Quindi non capisco proprio adesso quale sia il problema.

**Reply by M0ntyC0r4** [reply]
sulla plancia lo devi aggiungere te, cliccando la matita in alto a destra e poi "add card" in basso a sinistra, avrai la possibilità di scegliere quale card aggiungere alla plancia. a quel punto sceglierai "elementi immagine" e da lì potrai selezionare l'entità della telecamera che hai aggiunto.

Un altro modo è andare in impostazioni--dispositivi e servizi e da lì trovare l'integrazione arlo e cliccare sui dispositivi che segna come configurati, per poi fare aggiungi alla plancia e selezionare la dashboard corretta.

**Reply by JDrillz** [reply]
Non dovrebbe comparire in automatico nella panoramica?

**Reply by KC-inDomus** [reply]
aridaje. senza `camera`, cosa vuoi che ti mostri? :)

**Reply by JDrillz** [reply]
Sì ok, e quindi? Che dovrei fare per far venir fuori questo camera.* che come ho detto non ho?
Io ho seguito le indicazioni di un video australiano e copiato pari pari il codice sostituendo solo i dati di login, lui nel video non fa assolutamente niente di più, aggiunge il codice, si sposta sulla plancia e vede già i riquadri delle telecamere.

**Reply by KC-inDomus** [reply]
e io che ne so :) devi chiedere sul github dello sviluppatore, penso sia [questo](https://github.com/twrecked/hass-aarlo/issues). certo che se ti basi sui video youtube per capire e configurare la tua domotica, auguri...

**Reply by JDrillz** [reply]
Scusa se non sono nato illuminato come te, una volta si diceva "domandare è lecito, rispondere è cortesia", ma evidentemente la superbia supera la seconda in questo caso. Nessuno obbliga a rispondere.
Un giorno poi mi spiegherai il perché una cosa spiegata a video dovrebbe avere meno valore della stessa cosa scritta, magari persino dalla stessa persona, su un articolo, un forum o un libro (che sicuramente sarà la fonte principale di chi ha imparato ad usare HA immagino).

Grazie lo stesso ad entrambi

**Reply by M0ntyC0r4** [reply]
@KC-inDomus ma la fai finita di rispondere di merda a tutti?!?!
Non è che tutti sono disposti a sopportare il fatto che tu sei incazzato con il mondo perché ce l'hai piccolo eh..

**Reply by KC-inDomus** [reply]
> Un giorno poi mi spiegherai il perché una cosa spiegata a video dovrebbe avere meno valore della stessa cosa scritta, magari persino dalla stessa persona, su un articolo, un forum o un libro (che sicuramente sarà la fonte principale di chi ha imparato ad usare HA immagino).

te lo spiego facile: perché su youtube le cose spiegate invecchiano un minuto dopo che l'hai pubblicate - e te lo dimostra il fatto che quello che hai fatto non funziona. lo scritto si corregge, cosa che noi facciamo di continuo, i video no.
auguri 👍

**Comment by KC-inDomus** [comment]
> Cosa sbaglio? Devo fare altro per vedere qualcosa? Grazie.

hai controllato i log? l'integrazione viene caricata o causa errori? l'entità `camera.*` eventuale viene creata oppure no?
questo devi verificare, prima di aspettarti di leggere alcunché sulla dashboard.

**Reply by JDrillz** [reply]
Fra le integrazioni non compare perché è gestibile solo aggiungendolo manualmente al configuration.yaml, invece su dispositivi ho **arlo camera support**. Errori all'avvio di HA non me ne da. All'inizio sì, poi dopo aver sistemato quella storia del 2FA non segnala più niente di anomalo.

Fra le entità non compare nessun camera.* però ho un **update.arlo_camera_support_update** ed un **sensor.aarlo_total_cameras**

**Reply by M0ntyC0r4** [reply]
quand'è così puzza di camera che non ha attivo l'rtsp, sai?!?!
hai controllato su ispycam che la tua cam supporti l'rtsp?

**Reply by M0ntyC0r4** [reply]
ma sei sicuro che dopo av er installato il custom component da hacs tu non debba andare in impostazioni--dispositivi ed aggiungere l'integrazione??

**Reply by JDrillz** [reply]
Penso che tu abbia centrato il punto. 
Mi sono informato un po', e pare che non tutti i modelli di telecamere Arlo supportino l'RTSP, molto pochi lo fanno a dire il vero. 
Fra questi ovviamente non rientra la mia. 
Giusto per completezza di informazioni sì, avevo controllato che non dovessi aggiungere l'integrazione da impostazioni>dispositivi e mi diceva proprio che non era possibile farlo dall'interfaccia utente ma solo da yaml.

Ti ringrazio, anche se non è un problema risolvibile almeno non perderò altro tempo a fare tentativi vani e so quale marchio evitare la prossima volta.

**Reply by M0ntyC0r4** [reply]
Controlla su ispycam.
Io ho comprato lo spioncino digitale ezviz DP2C e ispycam diceva che era compatibile, poi quando l'ho provato non c'era il supporto RTSP...indagando un po', sono riuscito a capire che dovevo fare un downgrade del firmware, perché ezviz con gli aggiornamenti aveva disabilitato quella funzionalità, che invece prima c'era.

Tutto questo per dirti di dare uno sguardo in giro per vedere se magari c'è un modo, facendo un downgrade del firmware o installandone uno diverso (come si fa con i sonoff e il tasmota), di riuscire ad avere l'rtsp.

Prova pure a vedere se tante volte con l'addon di ha motioneye riesci invece ad avere un flusso video..

**Reply by JDrillz** [reply]
Controllato, sul database non c'è neanche la marca Arlo, ho provato pure motioneye ma niente.
Ho letto una risposta dell'assistenza di Arlo ad un ragazzo che chiede proprio dell'RTSP sulla mia telecamera e dicono "supporta i protocolli HTTP (port 80), HTTPS (port 443) e NTP (port 123). Il web client Arlo sfrutta uno streaming dinamico adattivo su HTTP (DASH o MPEG-DASH) per visualizzare i contenuti in streaming di tutte le camere Arlo in un browser".

Continuerò a dare un po' un'occhiata in giro sulla base di quanto mi hai suggerito, per vedere se qualcuno ha messo le mani sul firmware, ma la vedo bigia. Per il momento ti ringrazio di nuovo per gli spunti.

**Reply by M0ntyC0r4** [reply]
RTSPtoWebRTC?!?!
Non so bene cosa sia perché me l'ha suggerita l'altro giorno un utente qua ma non mi è sembrata la soluzione ottimale per me...però potrebbe esserlo per te.
Da quello che capisco ti permette proprio di vedere lo streaming rtsp delle cam in un browser...è un'integrazione di home assistant.
Vai a impostazioni--dispositivi--aggiungi integrazione, la cerchi e la trovi lì.
Puoi provare con quella...sennò devi rimandare le cam, sempre che tu le abbia prese con prime, allora hai il reso per un mese!!!!

**Comment by KC-inDomus** [comment]
> Cosa sbaglio? Devo fare altro per vedere qualcosa? Grazie.

Le Arlo si integrano benissimo tramite custom component. È tutto spiegato [qui](https://github.com/twrecked/hass-aarlo?tab=readme-ov-file). Non devi usare il componente ufficiale ma questo, e semplicemente configurarlo come spiegato nel link di cui sopra.

**Reply by M0ntyC0r4** [reply]
invece di perdere tempo a rispondergli di merda, potevi uscirtene 5 giorni fa, quando ha fatto la domanda.

**Reply by JDrillz** [reply]
Grazie, ad ogni modo haas-arlo è ciò che ho utilizzato fin dall'inizio. Quando nel primo messaggio ho scritto "Ho installato HACS, ho scaricato Arlo Camera Support" mi riferivo proprio ad hass-aarlo il cui nome una volta scaricato è proprio "Arlo Camera Support".
Comunque sia ho una teoria sul fatto che non stia funzionando la condivisione della telecamera, perciò ho già contattato direttamente Arlo, attendo risposta.
